File: src/main/java/com/bombproj/dao/UserDao.java
package com.bombproj.dao;

import com.bombproj.constants.UserState;
import com.bombproj.model.User;
import com.bombproj.vo.UserQueryResultVO;
import com.queryflow.accessor.A;
import org.springframework.stereotype.Repository;

import java.util.List;

@Repository
public class UserDao {

    public List<User> queryUserByNameOrEmailOrPhone(String userNameOrEmailOrPhone) {
        String sql = "SELECT * FROM users WHERE state = ? AND userName = ?" +
            "UNION SELECT * FROM users WHERE state = ? AND email = ?" +
            "UNION SELECT * FROM users WHERE state = ? AND phone = ?";
        return A.query(sql, UserState.VALID.getCode(), userNameOrEmailOrPhone,
            UserState.VALID.getCode(), userNameOrEmailOrPhone,
            UserState.VALID.getCode(), userNameOrEmailOrPhone).list(User.class);
    }

    public Integer queryUserCountByUsername(String userName) {
        String sql = "SELECT COUNT(id) FROM users WHERE state = ? AND userName = ?";
        return A.query(sql, UserState.VALID.getCode(), userName).one(Integer.class);
    }

    public Integer queryUserCountByEmail(String email) {
        String sql = "SELECT COUNT(id) FROM users WHERE state = ? AND email = ?";
        return A.query(sql, UserState.VALID.getCode(), email).one(Integer.class);
    }

    public Integer queryUserCountByPhone(String phone) {
        String sql = "SELECT COUNT(id) FROM users WHERE state = ? AND phone = ?";
        return A.query(sql, UserState.VALID.getCode(), phone).one(Integer.class);
    }

}

File: src/main/scala/com/bombproj/dao/UserDao.scala
package com.bombproj.dao

import java.util

import com.bombproj.constants.UserState
import com.bombproj.model.User
import com.queryflow.accessor.A
import org.springframework.stereotype.Repository

@Repository
class UserDao {

    def queryUserByNameOrEmailOrPhone(userNameOrEmailOrPhone: String): util.List[User] = {
        val sql = "SELECT * FROM users WHERE state = ? AND userName = ?" +
            "UNION SELECT * FROM users WHERE state = ? AND email = ?" +
            "UNION SELECT * FROM users WHERE state = ? AND phone = ?"
        A.query(sql, UserState.VALID.getCode.asInstanceOf[Integer], userNameOrEmailOrPhone,
            UserState.VALID.getCode.asInstanceOf[Integer], userNameOrEmailOrPhone,
            UserState.VALID.getCode.asInstanceOf[Integer], userNameOrEmailOrPhone)
            .list(classOf[User])
    }

    def queryUserCountByUsername(userName: String): Integer = {
        val sql = "SELECT COUNT(id) FROM users WHERE state = ? AND userName = ?"
        A.query(sql, UserState.VALID.getCode.asInstanceOf[Integer], userName).one(classOf[Integer])
    }

    def queryUserCountByEmail(email: String): Integer = {
        val sql = "SELECT COUNT(id) FROM users WHERE state = ? AND email = ?"
        A.query(sql, UserState.VALID.getCode.asInstanceOf[Integer], email).one(classOf[Integer])
    }

    def queryUserCountByPhone(phone: String): Integer = {
        val sql = "SELECT COUNT(id) FROM users WHERE state = ? AND phone = ?"
        A.query(sql, UserState.VALID.getCode.asInstanceOf[Integer], phone).one(classOf[Integer])
    }

}
